src/core/lib/event_engine/ares_resolver.cc

```diff
@@ -189,46 +188,82 @@ struct HostnameQueryArg : public QueryArg {
   std::vector<EventEngine::ResolvedAddress> result;
 };
 
-}  // namespace
-
-absl::StatusOr<grpc_core::OrphanablePtr<AresResolver>>
-AresResolver::CreateAresResolver(
-    absl::string_view dns_server,
-    std::unique_ptr<GrpcPolledFdFactory> polled_fd_factory,
-    std::shared_ptr<EventEngine> event_engine) {
+absl::Status InitAresChannel(
+    ares_channel* channel, absl::string_view dns_server,
+    const std::unique_ptr<GrpcPolledFdFactory>& polled_fd_factory) {
```

Comment:
I suggest passing this in as `const GrpcPolledFdFactory&`.  Exposing the fact that the caller happens to have a unique_ptr is irrelevant to this function.